tools/ctl/ctl.c

```diff
@@ -243,7 +243,8 @@ static void csv_data_dump(struct ctl_data *ctl_data, FILE *fh)
 	int s = 0;
 
 	config = &ctl_data->buffer[BUFFER_ABI_OFFSET];
-	n = ctl_data->buffer[BUFFER_SIZE_OFFSET] / sizeof(uint32_t);
+	n = ctl_data->buffer[BUFFER_SIZE_OFFSET] / sizeof(uint32_t) -
+		sizeof(struct sof_abi_hdr) / sizeof(uint32_t);
```

Comment:
I think this change is wrong. When I read from a real SOF device the control, the entire blob is not printed, the end is cut.

Reply:
hmm I suppose I could use the -r option to get around this. I will drop this change